test/e2e/network/service.go

```diff
@@ -2947,6 +2948,118 @@ var _ = SIGDescribe("Services", func() {
 		_, err = f.ClientSet.CoreV1().Endpoints(testNamespaceName).Get(context.TODO(), testEndpointName, metav1.GetOptions{})
 		framework.ExpectError(err, "should not be able to fetch Endpoint")
 	})
+
+	ginkgo.It("should complete a service status lifecycle", func() {
+
+		dc := f.DynamicClient
+		ns := f.Namespace.Name
+		svcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}
+		testServiceName := "test-service"
+		svcWatchTimeoutSeconds := int64(180)
+
+		ginkgo.By("creating a Service")
+		testService := v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   testServiceName,
+				Labels: map[string]string{"test-service-static": "true"},
+			},
+			Spec: v1.ServiceSpec{
+				Type: "ClusterIP",
+				Ports: []v1.ServicePort{{
+					Name:       "http",
+					Protocol:   v1.ProtocolTCP,
+					Port:       int32(80),
+					TargetPort: intstr.FromInt(80),
+				}},
+			},
+		}
+		_, err := cs.CoreV1().Services(ns).Create(context.TODO(), &testService, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Service")
+
+		ginkgo.By("watching for the Service to be added")
+		svcWatch, err := cs.CoreV1().Services(ns).Watch(context.TODO(), metav1.ListOptions{LabelSelector: "test-service-static=true", TimeoutSeconds: &svcWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "failed to create service")
+
+		svcWatchChan := svcWatch.ResultChan()
+		for event := range svcWatchChan {
```

Comment:
what if  status is still pending(adding)?  will this test no exist?

Reply:
After chatting with @BobyMCbobs I've updated the test to use some better processes around watches. 

@JornShen Does the current test process address your concerns?

Reply:
@heyste yes, impl is good.